ee/tabby-webserver/src/repositories/resolve.rs

```diff
@@ -284,15 +288,18 @@ impl RepositoryCache {
         Ok(resp)
     }
 
-    pub fn find_repository(&self, name: &str) -> Option<RepositoryConfig> {
-        let repository_lookup = self.repository_lookup.read().unwrap();
-        let key = repository_lookup
-            .keys()
-            .find(|repo| repo.repo_name == name)?;
-        let value = repository_lookup.get(key)?;
-        Some(RepositoryConfig::new_named(
-            key.repo_name.clone(),
-            value.git_url.clone(),
-        ))
+    pub fn find_repository(&self, name: &str) -> Result<RepositoryConfig> {
+        for entry in self.cache.bucket()?.iter() {
```

Comment:
iterating the whole bucket to map repository name to RepositoryConfig is obviously too expensive.

Why not use `RepositoryService.get_repository_by_name`?